docs/advanced/nat.md

```diff
@@ -1,14 +1,59 @@
+![The provider is behind a NAT](../_static/images/advanced/nat/provider-nat.svg)
+
+By default using the `peer` command, the gateway server is configured on the provider cluster.
+However, there may be cases where the provider cluster's nodes are not directly reachable, such as when they are behind a NAT, while the consumer cluster is directly accessible.
+For instance, in the image above, cluster 2 is behind a NAT and is therefore not directly reachable.
+
+To fix this issue, we can swap the roles of the gateways, configuring the client on the cluster provider and the server on the consumer.
+To do so, you need to use [manual peering](./manual-peering.md), setting the inter-cluster network up separately.
+
+![The gateway server has been on the consumer side](../_static/images/advanced/nat/consumer-nat.svg)
+
+In this case, once the gateway server has been moved to cluster 1, we are able to configure the Liqo networking without the need to configure port-mapping on the NAT, as the gateway server is directly reachable.
+
+In the case you would like to keep the gateway server on the provider, you can follow the instructions on the next paragraph.
+
+## Option B: configure the gateway server to be reachable through the NAT (or physical load balancer)
+
+The solution presented in this section allows to reach the gateway server even when it is behind a NAT (or, for example, a physical load balancer) which prevents it from being directly reachable.
+
+![The provider is behind a NAT](../_static/images/advanced/nat/port-address-override.svg)
+
+In this case, **you must configure port-mapping on the NAT** so that the traffic directed on a specific port of the NAT can be forwarded to one of nodes of the cluster.
+
+**When network is configured via `liqoctl` commands**, if the service is of type `NodePort`, Liqo automatically uses the external IPs of the nodes to configure the gateway client.
+
+You can force the Liqo gateway client to use the public IP or FQDN and the port of the NAT where we configured port mapping, by providing some options to `liqoctl peer` or `liqoctl network connect` commands when using `liqoctl`.
+[With declarative peering](./peering/peering-via-cr.md#configuring-the-client-gateway-consumer-cluster) you can directly provide these values in the `GatewayClient` manifest.
+
+For example, let's imagine we expose the gateway server with a `NodePort` on port `30742` and we configure port mapping on the NAT so that port 40582 of the NAT redirects to port `30742` of a node of the cluster.
+
+Of course, the client, running on cluster 1 cannot directly reach the gateway, but it needs to pass through the NAT.
+To do so, you need to pass the IP address or FQDN of the NAT and port where configured port mapping (`203.0.113.8` and `40582` in this case):
+
+```bash
+liqoctl peer --remote-kubeconfig $PATH_TO_CLUSTER2_KUBECONFIG --client-address $NAT_PUBBLIC_ADDR --client-port $NAT_MAPPING_PORT --server-service-nodeport $GATEWAY_SERVER_NODEPORT
+```
```

Comment:
I fell the explanation below may is rather verbose. I would suggest to write something like:

In our precise case, the above command line will become the following:
```bash
liqoctl peer --remote-kubeconfig $PATH_TO_CLUSTER2_KUBECONFIG --client-address 203.0.113.8 --client-port 40582 --server-service-nodeport 30742
```
From this syntax, I have some concerns about the terminology used, namely "client" and "server". Personally, I was more in favor of something like:

```bash
liqoctl peer --remote-kubeconfig $PATH_TO_CLUSTER2_KUBECONFIG --public-address 203.0.113.8 --public-port 40582 --server-service-nodeport 30742
```
(instead of "public" we can also use "external")

Reply:
First part of the comment fixed.
About the syntax, the names of these flags have been a little bit controversed. The problem here is that, with the peer/connect command you set up both the client and the server of the tunnel. So, a user, reading "public" or "external", might think he is configuring how the server is exposed. I believe that the message that should pass here is that with those flags you are overriding address and port used by the client.

Reply:
I understand your point, but the fact that you're "overriding" a value is just in the mind of who defined the command, not in the user's mind :-) 
The parameter says "--client-address" and "--client-port", which to me (as a user) I feel misleading. The client address is the internal one, not the external one.
This would be much more clear if you say "--override-client-address" and "--override-client-port", just in case you can accept a longer command.